scripts/benchmark/utils/chain_plumbing.py

```diff
@@ -0,0 +1,131 @@
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    NamedTuple,
+    Type
+)
+
+from eth_keys import (
+    keys
+)
+
+from eth_utils import (
+    decode_hex,
+    to_wei,
+)
+
+from eth_typing import (
+    Address
+)
+
+from evm import (
+    constants,
+    Chain
+)
+from evm.vm.base import (
+    BaseVM
+)
+from evm.chains.mainnet import (
+    MainnetChain
+)
+from evm.db.backends.memory import (
+    MemoryDB
+)
+
+AddressSetup = NamedTuple('AddressSetup', [
+    ('address', Address),
+    ('balance', int),
+    ('code', bytes)
+])
+
+ALL_VM = [vm for _, vm in MainnetChain.vm_configuration]
+
+FUNDED_ADDRESS_PRIVATE_KEY = keys.PrivateKey(
+    decode_hex('0x45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8')
+)
+
+FUNDED_ADDRESS = Address(FUNDED_ADDRESS_PRIVATE_KEY.public_key.to_canonical_address())
+
+DEFAULT_INITIAL_BALANCE = to_wei(1000, 'ether')
+
+SECOND_EXISTING_ADDRESS = Address(b'\0' * 19 + b'\x02')
+
+NON_EXISTING_ADDRESS = Address(b'\0' * 19 + b'\x03')
+
+GENESIS_PARAMS = {
+    'parent_hash': constants.GENESIS_PARENT_HASH,
+    'uncles_hash': constants.EMPTY_UNCLE_HASH,
+    'coinbase': constants.ZERO_ADDRESS,
+    'transaction_root': constants.BLANK_ROOT_HASH,
+    'receipt_root': constants.BLANK_ROOT_HASH,
+    'difficulty': 1,
+    'block_number': constants.GENESIS_BLOCK_NUMBER,
+    'gas_limit': constants.GENESIS_GAS_LIMIT,
+    'extra_data': constants.GENESIS_EXTRA_DATA,
+    'nonce': constants.GENESIS_NONCE
+}
+
+
+def genesis_state(setup: Iterable[AddressSetup]) -> Any:
+
+    state = {}
+
+    for value in setup:
+        state[value.address] = {
+            "balance": value.balance,
+            "nonce": 0,
+            "code": value.code,
+            "storage": {}
+        }
+
+    return state
+
+
+def chain_without_pow(
```

Comment:
Can you open up an issue to migrate these to use pre-generated fixtures so that we can be sure to capture the overhead caused by POW validation/verification.

Reply:
I updated the original ticket and propose we keep that open as tracking ticket until we have worked through the entire list of things that we want to get done regarding the benchmarking.

https://github.com/ethereum/py-evm/issues/592